examples/java/spring-boot/src/test/java/uk/gov/api/springboot/infrastructure/ScheduleHandlerTest.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.api.springboot.infrastructure;
+
+import static com.github.valfirst.slf4jtest.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.github.valfirst.slf4jtest.LoggingEvent;
+import com.github.valfirst.slf4jtest.TestLogger;
+import com.github.valfirst.slf4jtest.TestLoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.api.springboot.domain.model.Api;
+import uk.gov.api.springboot.domain.model.Registry;
+import uk.gov.api.springboot.domain.model.RegistryEntry;
+import uk.gov.api.springboot.domain.model.repositories.ApiStorage;
+import uk.gov.api.springboot.domain.services.FetcherService;
+
+@ExtendWith(MockitoExtension.class)
+class ScheduleHandlerTest {
+
+  @Mock private Registry registry;
+  @Mock private ApiStorage storage;
+  @Mock private FetcherService fetcherService;
+  @InjectMocks private ScheduleHandler handler;
+
+  @Nested
+  class Log {
+
+    private final TestLogger logger = TestLoggerFactory.getTestLogger(ScheduleHandler.class);
+
+    @AfterEach
+    void tearDown() {
+      logger.clear();
+    }
+
+    @Test
+    void messageIsLoggedWhenScheduledTaskIsStarted() {
+      handler.fetchAndSaveApis();
+
+      assertThat(logger).hasLogged(LoggingEvent.info("Fetching and saving APIS"));
+    }
+
+    @Test
+    void messageIsLoggedWhenScheduledTaskIsFinished() {
+      handler.fetchAndSaveApis();
+
+      assertThat(logger).hasLogged(LoggingEvent.info("Finished fetching and saving APIs"));
+    }
+  }
+
+  @Test
+  void registryEntriesArePassedToFetcher() {
+    var entry = new RegistryEntry("123", "https://api-endpoint.example");
+    var entries = new ArrayList<RegistryEntry>();
+    entries.add(entry);
+    when(registry.findAll()).thenReturn(entries);
+
+    handler.fetchAndSaveApis();
+
+    verify(fetcherService).fetch(entry);
+  }
+
+  @Test
+  void listOfFetchedApisAreSavedToStorage() {
+    var entry = new RegistryEntry("123", "https://api-endpoint.example-one");
+    var entries = new ArrayList<RegistryEntry>();
+    entries.add(entry);
+    when(registry.findAll()).thenReturn(entries);
```

Comment:
Any reason we've not done this instead?

```suggestion
    when(registry.findAll()).thenReturn(List.of(entry));
```